### DISCUSSION TITLE
Workflow for multiplexed samples prepared with the RACE protocol

### DISCUSSION THREAD

**Discussion by VicenteFR** [milaboratory/mixcr#1861] [General]
Hi,

We implemented your RNA-based 5'-RACE protocol with UMIs. For a long time, we have been using the full `MIGEC` workflow to preprocess the raw data, all the way from `Checkout` to `FilterCdrBlastResults`. Because `MIGEC` is not supported nowadays, we've decided to move on and implement `MIXCR` for the preprocessing and downstream analyses of this type of data. I appreciate you've nicely added a preset for this type of data,  `milab-human-rna-tcr-umi-race`, though it doesn't look like it supports demultiplexing of TCR samples based on master and slave barcodes, essentially the process that `MIGEC Checkout` would do.  Accordingly, I believe the following workflow should do for the analysis of this type of data:
1. `MIGEC Checkout` for TCR sample demultiplexing, dropping the `-u` option to avoid UMI extraction.
2. `MIXCR` with the `milab-human-rna-tcr-umi-race` preset applied to each properly demultiplexed TCR sample.
3. Downstream analyses (custom).

Do you spot any potential issues with this general workflow? Do you have any specific recommendations?

Last but not least, thanks so much for the development of such wonderful toolkits for the analysis of immune repertoires and thanks in advance for your kind input!

Sincerely,
Vicente

**Comment by mizraelson** [comment]
Hi Vicente,
MiXCR can demultiplex the data by samples if needed. What protocol do you use? The milab-human-rna-tcr-umi-race preset was designed for [this kit](https://milaboratories.com/human-tcr-rna-kit), in which the samples are marked with different Illumina indices and should therefore be present in different FASTQ files.

Sincerely,
Mark

**Comment by VicenteFR** [comment]
Thanks so much for your prompt response! Indeed, that's the protocol we use and I think you've clarified my questions. Thanks again!
